crates/bin/pcli/src/command/migrate.rs

```diff
@@ -119,6 +157,202 @@ impl MigrateCmd {
+                    .as_mut()
+                    .context("view service must be initialized")?
+                    .assets(AssetsRequest {
+                        filtered: false,
+                        include_specific_denominations: vec![],
+                        include_lp_nfts: true,
+                        include_delegation_tokens: true,
+                        include_unbonding_tokens: true,
+                        include_proposal_nfts: false,
+                        include_voting_receipt_tokens: false,
+                    })
+                    .await?;
+
+                // Build asset cache from the response
+                let mut asset_cache = penumbra_sdk_asset::asset::Cache::default();
+                let assets_stream = assets_response.into_inner();
+                let assets = assets_stream
+                    .try_collect::<Vec<_>>()
+                    .await
+                    .context("failed to collect assets")?;
+                for asset_response in assets {
+                    if let Some(denom) = asset_response.denom_metadata {
+                        let metadata =
+                            denom.try_into().context("failed to parse asset metadata")?;
+                        asset_cache.extend(std::iter::once(metadata));
+                    }
+                }
+
+                // Return all unspent notes from the view service
+                let all_notes = app
+                    .view
+                    .as_mut()
+                    .context("view service must be initialized")?
+                    .unspent_notes_by_account_and_asset()
+                    .await?;
+
+                // Track values per (subaccount, asset) for fee calculation
+                let mut subaccount_values: HashMap<(u32, asset::Id), Amount> = HashMap::new();
+
+                // Filter and spend notes only from subaccounts in the specified range
+                for (account, notes_by_asset) in all_notes {
+                    if from.contains(&account) {
+                        for notes in notes_by_asset.into_values() {
+                            for note in notes {
+                                let position = note.position;
+                                let note = note.note;
+                                let value = note.value();
+                                planner.spend(note, position);
+                                *subaccount_values
+                                    .entry((account, value.asset_id))
+                                    .or_default() += value.amount;
+                            }
+                        }
+                    }
+                }
+
+                if subaccount_values.is_empty() {
+                    anyhow::bail!("no notes found in the specified subaccount range");
+                }
+
+                // Find the subaccount with the most fee token to pay fees
+                let (&(fee_account, _), _) = subaccount_values
```

Comment:
Autodiscovery of an appropriate subaccount for being fees is slick; I would have hardcoded this as a CLI arg. Optionally, we could add a CLI arg here, but I love the default behavior doing the right thing.